### PR TITLE
feat: trait_variant::make supports rewriting of the original trait.

### PR DESCRIPTION
**Pull request by sargarass**
Hello, this PR should close the issue #18.

### CODE REVIEW

**Comment by SeaDve** [comment]
I think this could be made more flexible by doing something like:

```rust
#[trait_variant::rewrite(Send + Sync)]
pub trait SomeTrait {
    ...
}
```

**Comment by tmandry** [comment]
I see there's demand for this, but I too am in favor of something that more closely resembles the shape of the `make` macro. `#[trait_variant::rewrite(Send)]` is okay. Some alternatives to `rewrite`: `add`, `bound`

**Comment by sargarass** [comment]
> I see there's demand for this, but I too am in favor of something that more closely resembles the shape of the make macro. #[trait_variant::rewrite(Send)] is okay

@SeaDve, @tmandry made the update to accomplish this

**Comment by odysa** [comment]
I think it also closes my issue #29 
Thank you for your work.

**Comment by tmandry** [comment]
Thanks @sargarass, the implementation looks good.

As for the name, on Zulip @Jules-Bertholet [suggested](https://rust-lang.zulipchat.com/#narrow/stream/315482-t-compiler.2Fetc.2Fopaque-types) `trait_variant::require`, which I like even better than the options presented so far. What are your thoughts?

**Comment by sargarass** [comment]
@tmandry, Hmm, as for naming, I think that `trait_variant::rewrite` is much closer to `trait_variant::make` in the sense of 'creation,' 'generation,' etc., than `trait_variant::require`. Because of the similarities between them, I have not suggested to name it `trait_variant::with`, which also sounds good but doesn't look similar.

**Comment by tmandry** [comment]
The other suggestion by @traviscross was to allow `#[trait_variant::make(Send)]` to mean the same thing, which wouldn't require a new name. I think I'm pretty agnostic on that vs rewrite. 

When I get back from vacation next week I'll work on merging one of these two, unless someone else gets to it first.

**Comment by sargarass** [comment]
>  unless someone else gets to it first.

@tmandry well...

**Comment by traviscross** [comment]
Over in #30 I've implemented it using the `make(Send)` syntax.

Given the ongoing discussion about choosing an alternate name for a different macro, I've become less ambivalent and probably would prefer to just do it that way.

**Comment by sargarass** [comment]
> > unless someone else gets to it first.

> @tmandry well...

@traviscross Now we both made the same work :D

**Comment by sargarass** [comment]
> I've become less ambivalent and probably would prefer to just solve this use case with make.

As for the entire concept, I do support this idea

File: trait-variant/src/variant.rs [hunk 1]
```diff
@@ -104,21 +125,8 @@ fn mk_variant(attrs: &Attrs, tr: &ItemTrait) -> TokenStream {
     quote! { #variant }
 }
 
+// Transforms a one item declaration within the definition if it has `async fn` and/or `-> impl Trait` return types by adding new bounds.
```
Comment (tmandry) [inline trait-variant/src/variant.rs:128]:
```suggestion
// Transforms one item declaration within the definition if it has `async fn` and/or `-> impl Trait` return types by adding new bounds.
```

**Review by tmandry — APPROVED** [review]

**Comment by qrilka** [comment]
Is it planned to have this in a new release soon?
https://github.com/rust-lang/rust/pull/118257 adds warnings on nightly with the current release

**Pull request merged**